rust/src/treefile.rs

```diff
@@ -364,22 +362,19 @@ impl Treefile {
     }
 
     /// Do some upfront semantic checks we can do beyond just the type safety serde provides.
-    fn validate_config(config: &TreeComposeConfig) -> io::Result<()> {
+    fn validate_config(config: &TreeComposeConfig) -> Fallible<()> {
         // check add-files
         if let Some(files) = &config.add_files {
             for (_, dest) in files.iter() {
                 if !add_files_path_is_valid(&dest) {
-                    return Err(io::Error::new(
-                        io::ErrorKind::InvalidInput,
-                        format!("Unsupported path in add-files: {}", dest),
-                    ));
+                    bail!("Unsupported path in add-files: {}", dest);
```

Comment:
Since the other `InvalidInput` errors are kept, why change just this one?

Reply:
Yeah, this works. Just needed to add `.into()` to convert to a `failure::Error`.